test/e2e/antreapolicy_test.go

```diff
@@ -655,50 +655,36 @@ func testANPBasic(t *testing.T) {
 }
 
 // testAuditLoggingBasic tests that a audit log is generated when egress drop applied
-func testAuditLoggingBasic(t *testing.T) {
+func testAuditLoggingBasic(t *testing.T, data *TestData) {
 	builder := &ClusterNetworkPolicySpecBuilder{}
 	builder = builder.SetName("test-log-cnp-deny").
 		SetPriority(1.0).
-		SetAppliedToGroup(map[string]string{"pod": "a"}, nil, nil, nil)
+		SetAppliedToGroup(map[string]string{"pod": "a"}, map[string]string{"ns": "x"}, nil, nil)
 	builder.AddEgress(v1.ProtocolTCP, &p80, nil, nil, nil, map[string]string{"ns": "z"},
-		nil, nil, secv1alpha1.RuleActionDrop).
-		AddEgressLogging()
+		nil, nil, secv1alpha1.RuleActionDrop)
+	builder.AddEgressLogging()
 
-	reachability := NewReachability(allPods, true)
-	reachability.Expect(Pod("x/a"), Pod("z/a"), false)
-	reachability.Expect(Pod("x/a"), Pod("z/b"), false)
-	reachability.Expect(Pod("x/a"), Pod("z/c"), false)
-	reachability.Expect(Pod("y/a"), Pod("z/a"), false)
-	reachability.Expect(Pod("y/a"), Pod("z/b"), false)
-	reachability.Expect(Pod("y/a"), Pod("z/c"), false)
-	reachability.Expect(Pod("z/a"), Pod("z/b"), false)
-	reachability.Expect(Pod("z/a"), Pod("z/c"), false)
+	_, err := k8sUtils.CreateOrUpdateCNP(builder.Get())
+	failOnError(err, t)
 
-	testStep := []*TestStep{
-		{
-			"Port 80",
-			reachability,
-			[]metav1.Object{builder.Get()},
-			80,
-			0,
-		},
-	}
-	testCase := []*TestCase{
-		{"Audit Log CNP Drop Egress From All Pod:a to NS:z", testStep},
-	}
-	executeTests(t, testCase)
-}
+	// generate some traffic that will be dropped by test-log-cnp-deny
+	k8sUtils.Probe("x", "a", "z", "a", p80)
+	k8sUtils.Probe("x", "a", "z", "b", p80)
+	k8sUtils.Probe("x", "a", "z", "c", p80)
 
-// validateAuditLog validates whether testAuditLoggingBasic succeeded or failed
-func validateAuditLog() bool {
-	data, err := ioutil.ReadFile("/var/log/antrea/networkpolicy/np.log")
+	podXA, _ := k8sUtils.GetPod("x", "a")
+	nodeName := podXA.Spec.NodeName
+	antreaPodName, err := data.getAntreaPodOnNode(nodeName)
+	if err != nil {
+		t.Errorf("error occurred when trying to get the Antrea agent pod running on node %s: %v", nodeName, err)
+	}
+	cmd := []string{"cat", "/var/log/antrea/networkpolicy/np.log"}
+	stdout, _, err := data.runCommandFromPod(antreaNamespace, antreaPodName, "antrea-agent", cmd)
 	if err != nil {
-		return false
+		t.Errorf("error occurred when inspecting the audit log file: %v", err)
 	}
-	s := string(data)
-	// check whether s contains test CNP
-	fmt.Println(strings.Contains(s, "test-log-cnp-deny"))
-	return true
+	assert.Equalf(t, true, strings.Contains(stdout, "test-log-cnp-deny"), "audit log does not contain entries for test-log-cnp-deny")
```

Comment:
do you think we should add a `wait.Poll` with a short timeout here in case the log file is not updated right away, just to avoid potential flakiness in the future?

Reply:
Added the standard networkPolicyDelay here, should be enough time. If the log entry does not appear by then, it's going to be a real issue anyways.